src/services/common/serviceConfig.js

```diff
@@ -241,14 +241,28 @@ const axiosServiceCall = async (
       async response => {
         const updatedResponse = { ...response };
         const callbackResponse = serviceHelpers.memoClone(updatedResponse);
+        const updatedLocation = { url: undefined, config: undefined };
+
+        if (
+          !updatedConfig.poll.location ||
+          typeof updatedConfig.poll.location === 'string' ||
+          typeof updatedConfig.poll.location === 'function'
+        ) {
+          updatedLocation.url = updatedConfig.poll.location || updatedConfig.url;
+        }
+
+        if (updatedConfig.poll.location?.url) {
+          updatedLocation.url = updatedConfig.poll.location.url;
+          updatedLocation.config = updatedConfig.poll.location.config;
+        }
 
-        // passed config, allow future updates by passing modified poll config
+        // passed config, allow future updates by passing a modified poll config into a setTimeout
         const updatedPoll = {
           ...updatedConfig.poll,
-          // internal counter passed towards validate
-          __retryCount: updatedConfig.poll.__retryCount ?? 0,
-          // a url, or callback that returns a url to poll the put/posted url
-          location: updatedConfig.poll.location || updatedConfig.url,
+          // internal counter passed towards validate and status
+          __retryCount: updatedConfig.poll.__retryCount ?? -1,
+          // url, or function callback that returns a url string to poll the put/posted url
+          location: updatedLocation,
```

Comment:
required: annotation fails to mention optional object format for `location`. also confirm func annotation